client-app/src/desktop/tabs/other/routing/AdvancedRoutingPanel.tsx

```diff
@@ -0,0 +1,198 @@
+                            items: [Icon.info()],
+                            className: 'tb-sample-grid__selbar'
+                        })
+                    )
+                ),
+                tbar: [
+                    refreshButton(),
+                    colAutosizeButton(),
+                    span('Group by:'),
+                    select({
+                        bind: 'groupBy',
+                        options: [
+                            {value: 'city', label: 'City'},
+                            {value: 'trade_date', label: 'Trade Date'},
+                            {value: 'city,trade_date', label: 'City › Trade Date'},
+                            {value: null, label: 'None'}
+                        ],
+                        width: 160
+                    }),
+                    span('Sort by:'),
+                    select({
+                        bind: 'sortBy',
+                        options: [
+                            {value: 'id|desc', label: 'Company ID (Desc)'},
+                            {value: 'id|asc', label: 'Company ID (Asc)'},
+                            {value: 'company|desc', label: 'Company Name (Desc)'},
+                            {value: 'company|asc', label: 'Company Name (Asc)'},
+                            {value: 'city|desc', label: 'City (Desc)'},
+                            {value: 'city|asc', label: 'City (Asc)'},
+                            {value: 'trade_date|desc', label: 'Trade Date (Desc)'},
+                            {value: 'trade_date|asc', label: 'Trade Date (Asc)'},
+                            {value: null, label: 'None'}
+                        ]
+                    }),
+                    switchInput({
+                        label: 'Prevent Route Deactivation',
+                        onChange: () => (model.preventDeactivate = !model.preventDeactivate)
+                    }),
+                    filler(),
+                    vbox({})
+                ]
+            })
+        });
+    }
+});
+
+class AdvancedRoutingPanelModel extends HoistModel {
+    @observable groupBy = null;
+    @observable sortBy = null;
+    @observable preventDeactivate = false;
+
+    constructor() {
+        super();
+        makeObservable(this);
+
+        this.addReaction({
+            track: () => XH.routerState.params,
+            run: () => this.parseRouteParams(),
+            fireImmediately: true
+        });
+
+        this.addReaction({
+            track: () => [this.groupBy, this.sortBy, this.gridModel.selectedRecord?.id],
+            run: () => this.updateRoute(),
+            fireImmediately: true
+        });
+
+        window.addEventListener('beforeunload', e => {
+            if (!XH.routerState.name.startsWith('default.other.advancedRouting')) {
+                delete e.returnValue;
+                return;
+            }
+            if (this.preventDeactivate) e.preventDefault();
+        });
+    }
+
+    gridModel = new GridModel({
```

Comment:
We'd typically either set this within the ctor or (if very simple as it is) above